server/backend/cron/news.go

```diff
@@ -131,53 +135,90 @@ func (c *CronService) parseNewsFeed(source model.NewsSource) error {
 	return nil
 }
 
-// downloadAndSaveImage
-// Downloads an image from the url, converts it to 720p width to and saves it to the database.
-// Returns id of file in db, null int on error.
-func (c *CronService) downloadAndSaveImage(url string) null.Int {
+// getDatabaseIdForImageAndDownload
+// Returns a file id that was saved to the database if it already exists. Otherwise creates it and triggers download of the image
+func (c *CronService) getDatabaseIdForImageAndDownload(url string) (null.Int, error) {
+	targetFileName := fmt.Sprintf("%x.jpg", md5.Sum([]byte(url)))
+	var fileId null.Int
+	if err := c.db.Model(model.Files{}).Where("name = ?", targetFileName).Select("file").Scan(&fileId).Error; err != nil && err != gorm.ErrRecordNotFound {
+		log.Printf("Couldn't query database for file: %v", err)
+		return null.Int{}, err
+	}
+	if fileId.Valid { // file already in database -> return for current news.
+		return fileId, nil
+	}
+
+	// otherwise store in database:
+	file := model.Files{Name: targetFileName, Path: STORAGE_DIR + ImageDirectory}
+	if err := c.db.Transaction(func(tx *gorm.DB) error {
+		if err := tx.Create(&file).Error; err != nil {
+			log.Printf("Could not store new file to database: %v", err)
+			return err
+		}
+		return nil
+	}); err != nil {
+		return null.Int{}, err
+	}
+
+	go c.downloadFile(url, targetFileName, MAX_IMAGE_RETRYS)
```

Comment:
I'm wondering if this would exhaust the process / server RAM. If we add a new feed, it might have to download 100s of pictures. If the process dies mid download, we might never delete the file from the database and we have no way of knowing if the download worked.